### PR TITLE
Workaround for Option#orNull from Java

### LINKED ISSUE #11625: Option#orNull can't be used from java
I raised this issue before https://github.com/scala/bug/issues/10006, but the issue wasn't well enough specified, so it was closed.  Let me try again.

Scala options are painful to use from java, and java users often want to simply use the actual value or a null, which is often idiomatic in java.  In particular, tools like findbugs can even make this kind of programming somewhat pleasant.  For java compatibility, it's very useful to be able to convert directly from an `Option` type to a "raw value or null" type.  Option#orNull seems like it should be able to do this, but it can't.  The bug here to me is that you can't unwrap your Option easily.

Here was the error message I got at the time:

```
error] class AllOf implements RoutePredicate {
                       [error] blahblaaablhablah.java:24: method orNull in class scala.Option<A> cannot be applied to given types;
                       [error]   required: scala.Predef.$less$colon$less<scala.runtime.Null$,A1>
                       [error]   found: no arguments
                       [error]   reason: cannot infer type-variable(s) A1
                       [error]     (actual and formal argument lists differ in length)
                       [error]     String sanitizedPath = opt.orNull();
                       [error] Compile failed at Oct 24, 2016 1:38:55 PM [0.221s]

               compile(blabhlahblahblahbl/predicates:predicates) failed: Zinc compile failed.
```

Someone suggested that users try `Option#getOrElse(null)` which will compile, but will result in a NPE if the `else` case is ever taken.  The correct way of implementing it would be `Option#getOrElse(() -> null)`, but this is easy to get wrong, allocates a function unnecessarily, and is surprising.

How about adding a method like `Option#unsafeOrNull` that doesn't ask for implicit evidence, and directly returns the contents of the Option in the Some case, and null in the None case?

### PR DESCRIPTION
**Pull request by som-snytt**
Fixes scala/bug#11625

Just use conversion in test.

Remove inadvertent duplication in build.sbt.

### CODE REVIEW

**Review by lrytz — APPROVED** [review]
👍

File: build.sbt [hunk 1]
```diff
@@ -141,7 +141,6 @@ lazy val commonSettings = instanceSettings ++ clearSourceAndResourceDirectories
       Seq("-Xlint:-options")  // allow `-source 1.8` and `-target 1.8`
     else
       Seq()),
-  Compile / javacOptions ++= Seq("-g", "-source", "1.8", "-target", "1.8", "-Xlint:unchecked"),
```
Comment (SethTisue) [inline build.sbt:144]: why was this removed?

Comment (som-snytt) [reply build.sbt:144]: see a few lines earlier

Edit: I blinked at it a couple of times last night. I would be dismayed to learn that this was some sort of secret sbt handshake.

Comment (SethTisue) [reply build.sbt:144]: :+1:
